src/messages/authority.rs

```diff
@@ -6,37 +6,65 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use super::{RoutingMessage, VerifyStatus};
+use super::{DstLocation, Variant, VerifyStatus};
 use crate::{
     chain::{SectionKeyInfo, SectionProofSlice, TrustStatus},
-    crypto::signing::Signature,
+    crypto::signing::Signature as SimpleSignature,
     error::{Result, RoutingError},
-    id::PublicId,
+    id::{P2pNode, PublicId},
+    location::SrcLocation,
     xor_space::{Prefix, XorName},
+    ConnectionInfo,
 };
 use bincode::serialize;
-use serde::{Deserialize, Serialize};
-use std::fmt::{self, Debug, Formatter};
 
-#[derive(Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
-#[allow(clippy::large_enum_variant)]
+#[derive(Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
 pub enum SrcAuthority {
-    Section {
-        proof: SectionProofSlice,
-        signature: bls::Signature,
-    },
     Node {
         public_id: PublicId,
-        signature: Signature,
+        signature: SimpleSignature,
+    },
+    Section {
+        prefix: Prefix<XorName>,
+        signature: bls::Signature,
+        proof: SectionProofSlice,
     },
 }
 
 impl SrcAuthority {
+    pub fn location(&self) -> SrcLocation {
+        match self {
+            Self::Node { public_id, .. } => SrcLocation::Node(*public_id),
+            Self::Section { prefix, .. } => SrcLocation::Section(*prefix),
+        }
+    }
+
+    pub fn as_node(&self) -> Result<&PublicId> {
```

Comment:
Not 100% sure about this interface and `as_section`, they seem easy to use incorrectly.

Reply:
They simplify the message dispatch though. In what way could they be used incorrectly?